coreneuron/utils/randoms/nrnran123.cu

```diff
@@ -179,7 +179,7 @@ void nrnran123_set_globalindex(uint32_t gix) {
     // If the global seed is changing then we shouldn't have any active streams.
     {
         std::lock_guard<OMP_Mutex> _{g_instance_count_mutex};
-        if (g_instance_count != 0 && nrnmpi_myid == 0) {
+        if (g_instance_count != 0 && nrnmpi_myid == 0 && get_global_state().v[0] != gix) {
```

Comment:
Is stdout unacceptably verbose without this change?

When I added this warning then I didn't add this check because I thought it increased the chances of hiding bugs behind lucky choices of random seeds.

Reply:
The change is not really necessary because the item that was causing the warning has been fixed. ie use of a random123 instance to help with multisend two phase performance. The worry about "hiding bugs" is valid since in the absence of the multisend two phase fix, multiple psolve would have different transfer patterns, and I wouldn't have noticed that was happening. So I will revert line 182 above.

Reply:
By the way. On the NEURON side the old "nondeterminism" still occurs but it is perhaps not a bug since the transfer patterns are different only if the model changes, and not different  merely because psolve is called multiple times.
Nevertheless it might be reasonable to backport the multisend two phase random123 fix  from CoreNEURON to NEURON.